work_queue/src/work_queue.c

```diff
@@ -6315,9 +6317,13 @@ struct work_queue_task *work_queue_wait_internal(struct work_queue *q, int timeo
 	while( (stoptime == 0) || (time(0) < stoptime) ) {
 
 		BEGIN_ACCUM_TIME(q, time_internal);
-
+		// keep track of whether a task has been sent and recieved
+		int task_event = 0;
 		// task completed?
-		t = task_state_any(q, WORK_QUEUE_TASK_RETRIEVED);
+		if (t == NULL)
+		{
+			t = task_state_any(q, WORK_QUEUE_TASK_RETRIEVED);
+		}
 		if(t) {
```

Comment:
@David-Simonetti-ND Let's move this `if(t)` inside the `if(t == NULL)`. Otherwise the `change_task_state` and `task_failed++` may be executed several times for the same task.